src/sync-controller.js

```diff
@@ -45,54 +45,83 @@ export const syncPointStrategies = [
   //                    segment in the current timeline with timing data
   {
     name: 'Segment',
-    run: (syncController, playlist, duration, currentTimeline) => {
+    run: (syncController, playlist, duration, currentTimeline, currentTime) => {
       let segments = playlist.segments;
+      let syncPoint = null;
+      let lastDistance = null;
 
-      for (let i = segments.length - 1; i >= 0; i--) {
+      currentTime = currentTime || 0;
+
+      for (let i = 0; i < segments.length; i++) {
         let segment = segments[i];
 
         if (segment.timeline === currentTimeline &&
             typeof segment.start !== 'undefined') {
-          let syncPoint = {
-            time: segment.start,
-            segmentIndex: i
-          };
+          let distance = Math.abs(currentTime - segment.start);
+
+          // Once the distance begins to increase, we have passed
+          // currentTime and can stop looking for better candidates
+          if (lastDistance && lastDistance < distance) {
```

Comment:
Not two segments with the same start time, but if `currentTime` is equal to `segment.start` then `distance` is 0 and the `syncPoint` would be set to the following segment (if one exists) because when `lastDistance` is 0 for the next iteration, this condition is false and the next condition (where it checks for `!lastDistance`) would result in true (leading to an overwrite of syncPoint).